### ISSUE TITLE
Cannot generate a one-to-one relationship with the latest version of jhipster and jhipster-uml

### ISSUE THREAD

**Issue by andidev** [jhipster/jhipster-uml#40]
one-to-one entity creation now requires `otherEntityRelationshipName` and `otherEntityField` to be set when `ownerSide == true`

this is since https://github.com/jhipster/generator-jhipster/pull/1757

Sorry fogot to mention this when developing this PR.

One will get the following error trying to generate a one to one relationship with jhipster-uml
```
The entity Child is being created.
WARNING otherEntityRelationshipName is missing in .jhipster/Child.json for relationship with id 1, using child as fallback
ERROR otherEntityField is missing in .jhipster/Child.json for relationship with id 1
```

**Comment by andidev** [comment]
PR https://github.com/jhipster/generator-jhipster/pull/1796 solves this issue but it still needs to be implemented in jhipster-uml if a anyone wants to set a otherEntityField value other than id (this is possible for other relationships today, or am I wrong?)

**Comment by MathieuAA** [comment]
@CarlKlagba did all the work needed because of #34.
I'll look into it a bit later today.

**Comment by CarlKlagba** [comment]
This seems like only few changes. I will commit a fix when a can grab my laptop.

**Comment by CarlKlagba** [comment]
@andidev When generating an entity with a One-to-One not-owner relationship with java 7 there is this error:
[ERROR] /home/user/Documents/jhipsterproject/src/main/java/com/mycompany/myapp/web/rest/JobResource.java:[82,24] cannot find symbol
[ERROR] symbol:   method getSecond()
[ERROR] location: variable job of type com.mycompany.myapp.domain.Job

This ligne correspond to this ![ one ](https://github.com/jhipster/generator-jhipster/blob/9d67fc06a5ce691512038d89079e94ccdeab4ebc/entity/templates/src/main/java/package/web/rest/_EntityResource.java#L108) on the template _EntityResource.java.

I wonder what the "getSecond()" method is and how comes it hasn't been initialize.

**Comment by andidev** [comment]
Sorry, this is not good. I think that is my test data that leaked in to the code somehow probably cause of a copy paste action.

I'll fix it as soon as I can.

Thanks for noticing it.

**Comment by andidev** [comment]
Sorry I accidenty closed this. Fix for The bug should soon be merged. A bit sloppy cause Im bugfixing from my mobile lol

**Comment by andidev** [comment]
Fix for the jhipster bug is merged now with master branch

**Comment by CarlKlagba** [comment]
@andidev 
Sorry for the late response, I didn't had much time during the week end.
I have still a few mistakes, and I would like you to help me find where it comes from.

I still have compilation errors with the generated app in all the _EntityResource.java files:
1)
[ERROR] symbol:   variable HeaderUtil
[ERROR] location: class com.mycompany.myapp.web.rest.JobHistoryResource
[ERROR] /home/ippon/Documents/Applications/etudeUml/src/main/java/com/mycompany/myapp/web/rest/JobHistoryResource.java:[65,26] cannot find symbol

There is the import:
import com.mycompany.myapp.web.rest.util.HeaderUtil;
but not the java file.

 2)
[ERROR] symbol:   class ArrayList
[ERROR] location: class com.mycompany.myapp.web.rest.DepartementResource
[ERROR] /home/ippon/Documents/Applications/etudeUml/src/main/java/com/mycompany/myapp/web/rest/DepartementResource.java:[118,44] cannot find symbol

The ArrayList import is not done for some Entity that still uses it. 
The conditions to make the import are those one:
if (javaVersion == '7' && fieldsContainOneToOne == true && fieldsContainOwnerOneToOne == false) { %>
import java.util.ArrayList;<% } %>

I think that the problem comes from  the fact that the faulty entity, Departement, has both an One-to-One ownerside and One-to-One not-ownerside relationships  

3)
[ERROR] symbol:   variable filter
[ERROR] location: class com.mycompany.myapp.web.rest.EmployeeResource
[ERROR] /home/ippon/Documents/Applications/etudeUml/src/main/java/com/mycompany/myapp/web/rest/EmployeeResource.java:[79,34] cannot find symbol

the error seems to come from [this line](https://github.com/jhipster/generator-jhipster/blob/5fdd3ccd05298acda9f522758add07baabeb29d1/entity/templates/src/main/java/package/web/rest/_EntityResource.java#L108)



All the errors are related to One-to-One relationships and/or Java7 . I hope you have sometime to take a look a those issues, I will keep looking and update you if needed.

**Comment by andidev** [comment]
No problem. I dont have mutch time myself and I can only fix bugs from my mobile currently. I'll fix the bug on wendsday at latest.

Looks like you are right according 2. It is actually the same reason why error 3 occurs. I need to introduce a new in memoryvariable fieldsContainOneToOneWithOwnerFalse to fix it.

Problem 1 can probably be solved by rerunnung The JHipster generator since the file HeaderUtils seems to be new (since 13 days back)

**Comment by andidev** [comment]
@CarlKlagba 
Problem 2 and 3 should be fixed in jhipster/generator-jhipster#1813
I have tested this and it works now.

Problem 1 that you are having should be fixed by regenerating your jhipster app before running jhipster-uml as mentioned before. Hope it works this time.
